### ISSUE TITLE
ansi color support

### ISSUE THREAD

**Issue by clojurians-org** [jyp/dante#115]
i use stack + nix + dante for developing.

it display ansi character when some exception occurs. 
```
-- >>> :reload                                                                                                                                                                       
-- [1 of 1] Compiling Main             ( /Users/larluo/my-gitlab/my-work/hbenchmark/app/Main.hs, interpreted )                                                                       
-- ^[[;1m/Users/larluo/my-gitlab/my-work/hbenchmark/app/Main.hs:132:21-24: ^[[;1m^[[31merror:^[[0m^[[0m^[[;1m^[[0m^[[0m^[[;1m                                                        
--     • Couldn't match expected type ‘IO () -> IO b0’                                                                                                                               
--                   with actual type ‘[[EsTestRecord]]’                                                                                                                             
--     • In the second argument of ‘forM_’, namely ‘recs’                                                                                                                            
--       In a stmt of a 'do' block: forM_ (putES mgr) recs                                                                                                                           
--       In the expression:                                                                                                                                                          
--         do mgr <- mkManager                                                                                                                                                       
--            recs <- mkRecords "larluo3" 10 2000                                                                                                                                    
--            forM_ (putES mgr) recs^[[0m^[[0m                                                                                                                                       
-- ^[[0m^[[0m^[[0mFailed, no modules loaded.                                                                                                                                         
-- >>> :set -XOverloadedStrings                                                                                                                                                      
-- >>> main                                                                                                                                                                          
-- ^[[;1m<interactive>:12:2-5: ^[[;1m^[[31merror:^[[0m^[[0m^[[;1m^[[0m^[[0m^[[;1m                                                                                                    
--     • Variable not in scope: main                                                                                                                                                 
--     • Perhaps you meant ‘min’ (imported from Prelude)^[[0m^[[0m                                                                                                                   
-- ^[[0m^[[0m^[[0m                                                                                                                                                                   
                                                                                                                
```

i can add haskell-mode-hook to resolve it successfully.
```
(require 'ansi-color)
(defun display-ansi-colors ()
  (interactive)
  (ansi-color-apply-on-region (point-min) (point-max)))

(add-hook 'haskell-mode-hook 'display-ansi-colors)
```

but when i eval-block again, the ansi character become messay gain unless i active the function again manual.

**Comment by purcell** [comment]
The following patch probably fixes this, but I would note that my local Stack + GHC (8.6.5) combination doesn't output these ANSI control characters, so the actual issue might be with the command you've configured for `dante` to use.

```patch
diff --git a/dante.el b/dante.el
index f1d9121..fc8c7b4 100644
--- a/dante.el
+++ b/dante.el
@@ -40,6 +40,7 @@
 ;;; Code:
 
 (require 'cl-lib)
+(require 'ansi-color)
 (require 'dash)
 (require 'f)
 (require 'flycheck)
@@ -831,7 +832,7 @@ Calls DONE when done.  BLOCK-END is a marker for the end of the evaluation block
                        (or (and (search-forward-regexp "[ \t]*--[ \t]*\\([ \t]>>>\\|$\\)" block-end t 1)
                                 (match-beginning 0))
                            block-end)))
-      (insert (apply 'concat (--map (concat "-- " it "\n") (--remove (s-blank? it) (s-lines res)))))
+      (insert (apply 'concat (--map (concat "-- " it "\n") (--remove (s-blank? it) (s-lines (ansi-color-apply res))))))
       (beginning-of-line)
       (dante-eval-loop block-end))))
 
```

**Comment by jyp** [comment]
Should be fixed by 28e52ee49e71d1928ff5014ca175bc77842e998d
